lua/lspconfig/server_configurations/c3_lsp.lua

```diff
@@ -4,7 +4,14 @@ return {
   default_config = {
     cmd = { 'c3lsp' },
     root_dir = function(fname)
-      return util.root_pattern { 'project.json', '.git' }(fname)
+      --look for a project directory
+      local path = util.root_pattern { 'project.json', 'manifest.json' }(fname)
+      if path ~= nil then
+        return path
+      end
+
+      --look for .git directory
+      return util.find_git_ancestors(fname)
```

Comment:
```suggestion
      return util.root_patttern { 'project.json', 'manifest.json', '.git' }(fname)
```

Reply:
@glepnir will this also work if we are editing the branch inside the '.git' directory instead?

Reply:
this is an order will find every pattern until find root